### PR TITLE
signature hype

### PR DESCRIPTION
**Pull request by antonbobkov**

### CODE REVIEW

File: erlang/crypto_tests/rsa_private [hunk 1]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (antonbobkov) [inline erlang/crypto_tests/rsa_private]: IT'S A SECRET DON'T LOOK

File: erlang/crypto_tests/sign.erl~ [hunk 2]
```diff
@@ -0,0 +1,8 @@
+-module(hash).
+
+-export([cmd/0]).
+
+-include_lib("stdlib/include/assert.hrl").
+
+cmd() ->
+    pass.
```
Comment (antonbobkov) [inline erlang/crypto_tests/sign.erl~]: I should really edit my gitignore

File: erlang/crypto_tests/rsa_private (hunk 1 above)
Comment (pdlla) [reply erlang/crypto_tests/rsa_private]: not me you have to worry about 🤖🤖🤖

File: erlang/crypto_tests/sign.erl [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+read_key(private, Name) ->
+    {ok, PemBin} = file:read_file(Name),
+    [RSAEntry] = public_key:pem_decode(PemBin),
+    Key = public_key:pem_entry_decode(RSAEntry),
+    Key;
+
+read_key(public, Name) ->
+    {ok, Bin} = file:read_file(Name),
+    [{Key, _}] = public_key:ssh_decode(Bin, public_key),
+    % [{Key, _}] <- spent like literally an hour figuring that one out, so stupid ಠ_ಠ
```
Comment (pdlla) [inline erlang/crypto_tests/sign.erl]: D:

Comment (pdlla) [inline erlang/crypto_tests/sign.erl]: ༼ つ ◕_◕ ༽つ

File: erlang/crypto_tests/sign.erl [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+-include_lib("stdlib/include/assert.hrl").
+-include_lib("public_key/include/public_key.hrl"). 
+
+read_key(private, Name) ->
+    {ok, PemBin} = file:read_file(Name),
+    [RSAEntry] = public_key:pem_decode(PemBin),
+    Key = public_key:pem_entry_decode(RSAEntry),
+    Key;
+
```
Comment (pdlla) [inline erlang/crypto_tests/sign.erl]: maybe add type annotation comments DDD:

File: erlang/crypto_tests/sign.erl [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+read_key(public, Name) ->
+    {ok, Bin} = file:read_file(Name),
+    [{Key, _}] = public_key:ssh_decode(Bin, public_key),
+    % [{Key, _}] <- spent like literally an hour figuring that one out, so stupid ಠ_ಠ
+    Key.
+
+
+cmd() ->
```
Comment (pdlla) [inline erlang/crypto_tests/sign.erl]: move to unit test D:?

Comment (pdlla) [reply erlang/crypto_tests/sign.erl]: oh wait I see this is a unity test D:

File: erlang/crypto_tests/sign.erl [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    % HASH
+
+    Message = "hi",
+    BinMessage = list_to_binary(Message),
+    MessageHash = crypto:hash(sha256, BinMessage),
+    L = binary_to_list(MessageHash),
+    io:format("string ~p hash_sz: ~p hash: ~p ~n", [Message, length(L), L]),
+
+    % RSA PRIVATE
+    Key = read_key(private, "rsa_private"),
```
Comment (pdlla) [inline erlang/crypto_tests/sign.erl]: shouldn't this use my_crypto::read_file_key?

File: erlang/crypto_tests/sign.erl (hunk 4 above)
Comment (antonbobkov) [reply erlang/crypto_tests/sign.erl]: I'm just copy-pasting code - types are a blackbox

File: erlang/crypto_tests/sign.erl (hunk 5 above)
Comment (antonbobkov) [reply erlang/crypto_tests/sign.erl]: not really, just me messing with some code to see if it works

Comment (antonbobkov) [reply erlang/crypto_tests/sign.erl]: but yeah, i can make a unit test

File: erlang/crypto_tests/sign.erl (hunk 6 above)
Comment (antonbobkov) [reply erlang/crypto_tests/sign.erl]: unrelated file, i'll move it to a different folder

File: erlang/crypto_tests/sign.erl (hunk 4 above)
Comment (antonbobkov) [reply erlang/crypto_tests/sign.erl]: if it works it's good enough :P

Comment (antonbobkov) [reply erlang/crypto_tests/sign.erl]: also trust me knowing type doesn't make it better

3> Key = public_key:pem_entry_decode(RSAEntry, "abcd1234").
    #'RSAPrivateKey'{version = 'two-prime',
                 modulus = 1112355156729921663373...2737107,
                 publicExponent = 65537,
                 privateExponent = 58064406231183...2239766033,
                 prime1 = 11034766614656598484098...7326883017,
                 prime2 = 10080459293561036618240...77738643771,
                 exponent1 = 77928819327425934607...22152984217,
                 exponent2 = 36287623121853605733...20588523793,
                 coefficient = 924840412626098444...41820968343,
                 otherPrimeInfos = asn1_NOVALUE}

File: erlang/crypto_tests/sign.erl (hunk 6 above)
Comment (antonbobkov) [reply erlang/crypto_tests/sign.erl]: this was just me messing with some stuff before creating my_crypto library

File: erlang/crypto_tests/rsa_private (hunk 1 above)
Comment (antonbobkov) [reply erlang/crypto_tests/rsa_private]: wait until we forget and use this key in the actual game...

File: erlang/crypto_tests/sign.erl (hunk 6 above)
Comment (pdlla) [reply erlang/crypto_tests/sign.erl]: 👍👍👍

File: erlang/crypto_tests/sign.erl~ (hunk 2 above)
Comment (pdlla) [reply erlang/crypto_tests/sign.erl~]: 👍

**Pull request merged**